### PR TITLE
Remove calls to `Str::contains` and `Arr::get` when not necessary

### PR DESCRIPTION
**Pull request by GromNaN**

### CODE REVIEW

File: src/Connection.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ public function __construct(array $config)
         $dsn = $this->getDsn($config);
 
         // You can pass options directly to the MongoDB constructor
-        $options = Arr::get($config, 'options', []);
+        $options = $config['options'] ?? [];
```
Comment (GromNaN) [inline src/Connection.php:50]: This is exactly what was done in `Add::get`. https://github.com/laravel/framework/blob/256f4974a09e24170ceeeb9e573651fd5e1c703e/src/Illuminate/Collections/Arr.php#L322-L324

File: src/Eloquent/Model.php [hunk 2]
```diff
@@ -155,7 +155,7 @@ public function getAttribute($key)
         }
 
         // Dot notation support.
-        if (Str::contains($key, '.') && Arr::has($this->attributes, $key)) {
+        if (str_contains($key, '.') && Arr::has($this->attributes, $key)) {
```
Comment (GromNaN) [inline src/Eloquent/Model.php:158]: `$key` is a string.

Comment (jmikola) [reply src/Eloquent/Model.php:158]: Based on https://laravel.com/docs/10.x/helpers#method-str-contains, am I correct to assume that `Str::contains()` only differs in its handling of array parameters?

Is it feasible to use scalar type hints for the `getAttribute()` signature, or would that conflict with inheritance? Likewise for other `str_contains()` changes below.

File: src/Connection.php (hunk 1 above)
Comment (jmikola) [reply src/Connection.php:50]: Noted that we're not using dot notation here (per [`Arr::get()` docs](https://laravel.com/docs/10.x/helpers#method-array-get)), so these changes seem sensible.

File: src/Eloquent/Model.php [hunk 3]
```diff
@@ -195,7 +195,7 @@ public function setAttribute($key, $value)
 
             $value = $builder->convertKey($value);
         } // Support keys in dot notation.
-        elseif (Str::contains($key, '.')) {
+        elseif (str_contains($key, '.')) {
```
Comment (jmikola) [inline src/Eloquent/Model.php:198]: Where does this package require PHP 8+? I don't see it in `composer.json` so I reckon it's an indirect dependency. I think it'd make sense to explicitly require it if we're going to be utilizing PHP 8+ functionality directly.

Hhappy to defer to a subsequent PHPORM ticket.

**Review by jmikola — APPROVED** [review]
One question about adding `string` type hints but LGTM.

File: src/Eloquent/Model.php (hunk 3 above)
Comment (GromNaN) [reply src/Eloquent/Model.php:198]: It requires `laravel/framework: ^10.0` which [requires PHP 8.1+](https://github.com/laravel/framework/blob/10.x/composer.json#L18).

**Comment by GromNaN** [comment]
Moved to jenssegers/laravel-mongodb#2571

**Pull request closed without merging**